models/comments.model.js

```diff
@@ -14,20 +17,51 @@ exports.fetchCommentsByArticleId = (article_id) => {
       }
 
       const queryStr = `
-      SELECT comment_id, votes, created_at, author, body, article_id
-      FROM comments
-      WHERE article_id = $1
-      ORDER BY created_at DESC;
-    `;
+          SELECT comment_id, votes, created_at, author, body, article_id
+          FROM comments
+          WHERE article_id = $1
+          ORDER BY created_at DESC;
+        `;
+
+      return db
+        .query(queryStr, [articleId])
+        .then(({ rows: comments }) => comments);
+    });
+  });
+};
+
+exports.addCommentForArticle = (username, body, article_id) => {
+  return Promise.all([
+    validateArticleId(article_id),
+    validateUsername(username),
+    validateCommentBody(body),
+  ]).then(([articleId, validUsername, validBody]) => {
+    return Promise.all([
+      articleExists(articleId),
+      checkUserExists(validUsername),
```

Comment:
You dont need to check user manually. It looks like you may be doing it twice here 🤔